1.4/Source/VanillaPsycastsExpanded/HarmonyPatches/Transferable_CanAdjustBy_Patch.cs

```diff
@@ -2,18 +2,30 @@
 {
     using HarmonyLib;
     using RimWorld;
+    using System.Collections.Generic;
     using System.Diagnostics;
+    using System.Linq;
     using UnityEngine;
     using Verse;
 
     [HarmonyPatch(typeof(Transferable), "CanAdjustBy")]
 	public static class Transferable_CanAdjustBy_Patch
     {
         public static Transferable curTransferable;
+        private static readonly HashSet<ThingDef> eltexThings;
+
+        static Transferable_CanAdjustBy_Patch()
+        {
+            eltexThings = DefDatabase<RecipeDef>.AllDefs
+                .Where(recipe => recipe.ingredients.Any(x => x.IsFixedIngredient && x.FixedIngredient == VPE_DefOf.VPE_Eltex))
+                .Select(recipe => recipe.ProducedThingDef)
+                .ToHashSet();
+        }
```

Comment:
Could you put this in `PsycastUtility` instead? I think it makes more sense for a cache like this to be in the utility method instead of a patch. While you're at it, move the `IsEltexOrHasEltexMaterial` method as well.